pkg/runner/run_context.go

```diff
@@ -779,3 +784,37 @@ func (rc *RunContext) localCheckoutPath() (string, bool) {
 	}
 	return "", false
 }
+
+func (rc *RunContext) handleCredentials() (username, password string, err error) {
+	// TODO: remove below 2 lines when we can release act with breaking changes
+	username = rc.Config.Secrets["DOCKER_USERNAME"]
+	password = rc.Config.Secrets["DOCKER_PASSWORD"]
+
+	container := rc.Run.Job().Container()
+	if container == nil {
+		return
+	}
+
+	if len(container.Credentials) != 2 {
```

Comment:
```suggestion
	if container.Credentials != nil && len(container.Credentials) != 2 {
```
my bad, forgot to check stuff